TESTS/mbed_hal/common_tickers/main.cpp

```diff
@@ -54,6 +54,11 @@ extern "C" {
 
 #define MAX_FUNC_EXEC_TIME_US 20
 #define DELTA_FUNC_EXEC_TIME_US 5
+#if defined(__MICROLIB)
```

Comment:
we dont have small std rather but specific one (microlib) only?

Reply:
Microlib slow division causes this issue of slow execution and perfectly works with newlib-nano

Reply:
Would it be an option to disable microlib on this platform, pending resolution of the speed issue?

Reply:
Yes, we can do. @evedon @MarceloSalazar please comment on your view

Reply:
I'm not keen on adding specific target dependencies as the code becomes a bit of a mess and it's hard to maintain/scale.
But we can detect the situation that prevents code from being executed and skip accordingly (for now). Thoughts?

Reply:
I meant not support use of microlib on this target - I believe it can be flagged as unsupported in targets.json?

Reply:
Probably a question for @jeromecoutant on how he thinks this could/should be handled.

Reply:
I don't think I will "unsupport" some specific targets, before we understand the real reason!
Please correct first #12221

Reply:
Another approach would be to optimise `HAL_GetTick`. Once the OS is running, it is going via the generic `ticker_read_us`, which is a fully-generalised routine to get microseconds.

And I believe it's only ever used to get small timing delays. A suggestion would be to use `us_ticker_read` directly, as you do when `mbed_sdk_initted` is false. I don't see what the `if` condition is gaining you there - why can't you always use the "uninited" code?

I'm busy space/speed optimising `ticker_read_us`, but still, you can just bypass it.

And then the next trick is to optimise the division for locality. When speed matters, you know little time has elapsed. So divide the /difference/ by 1000, not the total. The standard divide routines /should/ terminate fast for small inputs, but if they don't you could code them yourself.

Something like

     uint32_t prev_time, remainder;
     uint32_t new_time = us_ticker_read();
     uint32_t elapsed_time = ((new_time - prev_time) & US_TICKER_MASK) + remainder;
     uint32_t elapsed_ticks = elapsed_time / 1000;
     // compiler should terminate fast. Probably worth helping it with
     // if (elapsed_time < 1000) { elapsed_ticks = 0; } else if (elapsed_time < 2000) { elapsed_ticks = 1; } else { elapsed_ticks = elapsed_ticks / 1000; }
     remainder = elapsed_time % 1000;
     // compiler should optimise the above line - if it doesn't
     // remainder = elapsed_time - elapsed_ticks * 1000;
     return prev_time += elapsed_ticks;

Reply:
As Kevin suggestion, HAL_GetTick API source has optimized to achieve better performance on all profile and test results on one-time execution as below
| Ticker API  | _HAL_GetTick_ **without** optimization on bare metal profile with small C lib |_HAL_GetTick_ **with** optimization on bare metal profile with small C lib|
| ------------- | ------------- | --------------------------|
| clear_interrupt  | 51us  | 5us |
| disable_interrupt  | 51us  | 5us|